### PR TITLE
ztp: Add ts2phc configuration to new ptpconfig 

### PR DESCRIPTION
**Pull request by aneeshkp**
ptp config to support ts2phc implementation

### CODE REVIEW

**Comment by yliu127** [comment]
lgtm

**Comment by josephdrichard** [comment]
Also needs the operatorconfig change to enable this.
```
apiVersion: ptp.openshift.io/v1
kind: PtpOperatorConfig
metadata:
  name: default
  namespace: openshift-ptp
  annotations:
    ran.openshift.io/ztp-deploy-wave: "10"
spec:
  daemonNodeSelector:
    node-role.kubernetes.io/$mcp: ""
  plugins:
  - e810
```

**Comment by yliu127** [comment]
@josephdrichard is the new  PtpOperatorConfig plugins spec  compatible with previous ptp operator versions? If not, we need to discuss with @imiller0  on the best approach. May add a new source-cr or add a commented spec as an example but let user to actually add it in their PGT.

**Comment by josephdrichard** [comment]
We should probably have a new PtpOperatorConfig for this, similar to what this commit is doing to create a new PtpConfig

File: ztp/source-crs/PtpConfigGmWpc.yaml [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+      transportSpecific 0x0
+      ptp_dst_mac 01:1B:19:00:00:00
+      p2p_dst_mac 01:80:C2:00:00:0E
+      udp_ttl 1
+      udp6_scope 0x0E
+      uds_address /var/run/ptp4l
+      #
+      # Default interface options
+      #
+      clock_type OC
```
Comment (josephdrichard) [inline ztp/source-crs/PtpConfigGmWpc.yaml:127]: Should this be BC?  I don't see any option for specifying GM here.

File: ztp/source-crs/PtpConfigGmWpc.yaml [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+      # Default Data Set
+      #
+      twoStepFlag 1
+      priority1 128
+      priority2 128
+      domainNumber 24
+      #utc_offset 37
+      clock_class_threshold 165
+      clockClass 248
+      clockAccuracy 0xFE
```
Comment (josephdrichard) [inline ztp/source-crs/PtpConfigGmWpc.yaml:57]: For E810, I believe this is within 50 μs, which would gfall in this category
0x27:  The time is accurate to within 100 μs

File: ztp/source-crs/PtpConfigGmWpc.yaml [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      [global]
+      #
+      # Default Data Set
+      #
+      twoStepFlag 1
+      priority1 128
+      priority2 128
+      domainNumber 24
+      #utc_offset 37
+      clock_class_threshold 165
```
Comment (josephdrichard) [inline ztp/source-crs/PtpConfigGmWpc.yaml:55]: Drop.  Not applicable for GM.

Comment (aneeshkp) [reply ztp/source-crs/PtpConfigGmWpc.yaml:55]: DOne

File: ztp/source-crs/PtpConfigGmWpc.yaml [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+      clockClass 248
```
[the remark this answers is not in the document]
Comment (aneeshkp) [reply ztp/source-crs/PtpConfigGmWpc.yaml:56]: Done

File: ztp/source-crs/PtpConfigGmWpc.yaml (hunk 2 above)
Comment (aneeshkp) [reply ztp/source-crs/PtpConfigGmWpc.yaml:57]: done

File: ztp/source-crs/PtpConfigGmWpc.yaml (hunk 1 above)
Comment (aneeshkp) [reply ztp/source-crs/PtpConfigGmWpc.yaml:127]: done

File: ztp/source-crs/PtpConfigGmWpc.yaml [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+      ingressLatency 0
+      boundary_clock_jbod 0
+      #
+      # Clock description
+      #
+      productDescription ;;
+      revisionData ;;
+      manufacturerIdentity 00:00:00
+      userDescription ;
+      timeSource 0xA0
```
Comment (josephdrichard) [inline ztp/source-crs/PtpConfigGmWpc.yaml:143]: timeSource should be 0x20 for GNSS

Comment (aneeshkp) [reply ztp/source-crs/PtpConfigGmWpc.yaml]: Fixed

**Review by josephdrichard — APPROVED** [review]
/lgtm

**Pull request merged**